File: backend/app/schemas/indicator.py
from datetime import datetime
from typing import Any

from pydantic import BaseModel, ConfigDict


class IndicatorValueBase(BaseModel):
    instrument_id: int
    indicator_name: str
    category: str
    timeframe: str
    timestamp: datetime
    values: dict[str, Any]


class IndicatorValueCreate(IndicatorValueBase):
    pass


class IndicatorValueRead(IndicatorValueBase):
    model_config = ConfigDict(from_attributes=True)

    id: int

File: frontend/src/types/indicator.ts
export type IndicatorPayloadValue = number | string | null;

export type IndicatorValue = {
  id: number;
  instrument_id: number;
  indicator_name: string;
  category: string;
  timeframe: string;
  timestamp: string;
  values: Record<string, IndicatorPayloadValue>;
};
